clarity/src/vm/analysis/type_checker/natives/mod.rs

```diff
@@ -555,9 +555,10 @@ fn check_get_block_info(
         .ok_or(CheckError::new(CheckErrors::GetBlockInfoExpectPropertyName))?;
 
     let block_info_prop =
-        BlockInfoProperty::lookup_by_name(block_info_prop_str).ok_or(CheckError::new(
-            CheckErrors::NoSuchBlockInfoProperty(block_info_prop_str.to_string()),
-        ))?;
+        BlockInfoProperty::lookup_by_name_at_version(block_info_prop_str, &checker.clarity_version)
```

Comment:
regarding this:

> I could go back and add schema migration code for these two things, but I'm hesitant to do so, since the time taken to write, test, and review the migration code will likely exceed the time to just boot up a 2.1 node from genesis.

Are you comparing the time of the miners to your time? I mean it's fine to make this tradeoff, but not for the reason that the users' time is 1:1 substitutable with yours.

Reply:
It takes less than 30 hours to boot from genesis in the latest release.  This is something you could kick off on a Friday and expect to be done on a Monday.

More generally, I think we want to get folks into the habit of spawning a node from genesis on each hard fork.  It would:

* reduce the codebase's technical debt by giving us a chance to remove the last epoch's schema migrations

* de-risk the chance that a bug in the ever-growing sequence of schema migrations leads to a chain split

* enable us to be able to make significant, principled, and even green-field changes to the database schemas without worrying about compatibility, thereby reducing development, review, and testing time for deploying them

I think it's fair to consider each new epoch of Stacks to be a wholly new blockchain, whose genesis state just happens to be derived from the last epoch's chainstate.  In fact, I'd be supportive of shipping each hard fork with all of the blocks from the last epochs as a database, so they don't even need to be downloaded on boot-up (just re-processed and re-validated against the Bitcoin blockchain).